databricks/koalas/tests/test_groupby.py

```diff
@@ -810,6 +810,15 @@ def test_idxmin(self):
         self.assert_eq(pdf.groupby(('x', 'a')).idxmin(skipna=False),
                        kdf.groupby(('x', 'a')).idxmin(skipna=False).sort_index())
 
+    def test_head(self):
+        pdf = pd.DataFrame({'a': [1, 1, 2, 2, 3],
+                            'b': [1, 2, 3, 4, 5],
+                            'c': [5, 4, 3, 2, 1]}, columns=['a', 'b', 'c'])
+        kdf = ks.from_pandas(pdf)
+
+        self.assert_eq(pdf.groupby(['a', 'b']).head(2),
+                       kdf.groupby(['a', 'b']).head(2).sort_index())
```

Comment:
Could you add more tests? e.g., multi-index, multi-index columns, tuple named Series, large n, small n, negative n, random order index, etc.

Reply:
it sounds good, i'll add 😄

Reply:
pandas seems doesn't support groupby for tuple-named Series like below.

```python
>>> pser = pd.Series([1, 2, 3, 4, 5], name=('i', 'am', 'a', 'series'))
>>> pser.groupby(pser.name).head(2)
KeyError: ('i', 'am', 'a', 'series')
```

but it works at koalas, but result is not correct.

```python
>>> kser = ks.Series([1, 2, 3, 4, 5], name=('i', 'am', 'a', 'series'))
>>> kser.groupby(kser.name).head(2)
4    5
0    1
2    3
1    2
3    4
Name: (i, am, a, series), dtype: int64
```

shall we make it work? or raises `KeyError` like pandas?

Reply:
It's not the correct usage anyway.

```py
>>> import pandas as pd
>>> pser = pd.Series([1, 2, 3, 4, 5], name='x')
>>> pser.groupby(pser.name).head(2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ueshin/workspace/databricks-koalas/miniconda/envs/databricks-koalas_3.6/lib/python3.6/site-packages/pandas/core/generic.py", line 7632, in groupby
    observed=observed, **kwargs)
  File "/Users/ueshin/workspace/databricks-koalas/miniconda/envs/databricks-koalas_3.6/lib/python3.6/site-packages/pandas/core/groupby/groupby.py", line 2110, in groupby
    return klass(obj, by, **kwds)
  File "/Users/ueshin/workspace/databricks-koalas/miniconda/envs/databricks-koalas_3.6/lib/python3.6/site-packages/pandas/core/groupby/groupby.py", line 360, in __init__
    mutated=self.mutated)
  File "/Users/ueshin/workspace/databricks-koalas/miniconda/envs/databricks-koalas_3.6/lib/python3.6/site-packages/pandas/core/groupby/grouper.py", line 578, in _get_grouper
    raise KeyError(gpr)
KeyError: 'x'
```

Koalas currently allows it but we might need to block it?

Reply:
@ueshin thanks for giving an opinion.

i also think we better block it also like pandas.

i'll fix them.

Reply:
i just open an issue(#1095 ) to handle this.

after quickly solve that, i'll come back here and go on.